File: src/main/controller/HomeController.java
package main.controller;

import main.MainDispatcher;
import main.model.User;
import main.service.UserService;

public class HomeController implements Controller {

    private UserService userService;

    public HomeController() {
        userService = new UserService();
    }

    public void doControl(Request request) {
        String type = "";
        User user = userService.getLoggedUser();
        String username = user.getUsername();
        if (request != null) {
            type = user.getType();
            if (type.equals("driver"))
                MainDispatcher.getInstance().callView("HomeDriver", request);
            else
                MainDispatcher.getInstance().callView("HomeOwner", request);
        } else {
            type = user.getType();
            request = new Request();
            request.put("username", username);
            if (type.equals("driver")) {
                MainDispatcher.getInstance().callView("HomeDriver", request);
            } else
                MainDispatcher.getInstance().callView("HomeOwner", request);
        }
    }
}

File: PCarpet/Controllers/HomeController.cs
using PCarpet.Service;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;

namespace PCarpet.Controllers
{
    public class HomeController : Controller
    {
        UserService userService;

        public HomeController()
        {
            userService = new UserService();
        }

        public ActionResult Index()
        {
            ViewBag.Title = "Home Page";

            return View();
        }

        public ActionResult Home(string username, string password)
        {
           
            using (pcarpetEntities a = new pcarpetEntities()) {

                //a.user.Add(new user("Cacca", "Pipi", 0, "3331231231", "dasdas@"));
                //a.SaveChanges();

                //user user = a.user.Find(username);
                Boolean login = userService.login(username, password);
                if (login)
                {
                    user user = userService.getLoggedUser();
                    int type = user.type;
                    if (type == 1)
                        return View("homeDriver");
                    else if (type == 0)
                        return View("homeOwner");
                    else if (type == 2)
                        return View("homeCop");
                }
                else
                {
                    ViewBag.feedback = "loginerror";
                    return View("index");
                }
                    



            }
            return View("HomeDriver");
        }
    }
}
